### PR TITLE
PIRK-45 Remove dependency on Java serialization

### PR DESCRIPTION
**Pull request by wraydulany**
This is a WIP (do not merge) for adding in JSON serialization. I can't get this WIP to pass the elasticsearch spark streaming tests on my cluster.

I'm fairly sure that this is a configuration problem on my end (the errors I get appear to be such), but I'm having little luck resolving them, and would love it if some of the community could give it a test.

I've left in Java Serialization for now, but I've made JSON serialization the default.

### CODE REVIEW

File: src/main/java/org/apache/pirk/querier/wideskies/QuerierDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+package org.apache.pirk.querier.wideskies;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.google.gson.*;
```
Comment (smarthi) [inline src/main/java/org/apache/pirk/querier/wideskies/QuerierDeserializer.java]: Please avoid using import *.

File: src/main/java/org/apache/pirk/query/wideskies/QueryDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+import org.apache.pirk.schema.query.filter.DataFilter;
+import org.apache.pirk.schema.query.filter.FilterFactory;
+import org.apache.pirk.utils.PIRException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.math.BigInteger;
+import java.util.*;
```
Comment (wraydulany) [inline src/main/java/org/apache/pirk/query/wideskies/QueryDeserializer.java]: Yeah, there are 8 of those. Fixing.

**Comment by wraydulany** [comment]
Fixed the `import...*` and also removed some code garbage that i'd commented out, from an earlier attempt to use jackson, an attempt killed by spark and its ancient version of jackson.

File: src/main/java/org/apache/pirk/query/wideskies/Query.java [hunk 3]
```diff
@@ -26,100 +26,103 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Consumer;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
```
Comment (ellisonanne) [inline src/main/java/org/apache/pirk/query/wideskies/Query.java]: Need to remove Jackson imports

**Comment by ellisonanne** [comment]
A couple of general comments - (1) need to remove the Jackson imports and dependency as it GSON is being used (2) need to format with the Pirk codestyle file.

I'll take a look at the ES issues...

**Comment by wraydulany** [comment]
Ok, I've expurgated references (import and comment) to jackson in the code, and I've run the style checker against the code i touched. 

Additionally, I've investigated the problems with the 1:ES and 1:ESS tests that I'm having, and I'm convinced that they are due to misconfiguration of the cluster I'm testing on.

This passes the build tests and distributed tests 1:J, 1:JS, 1:JSS, 1:SS, and 1:E for me. If anyone has been able to successfully execute the 1:ES and/or 1:ESS tests with this PR *and* fail them, please comment and kill this PR. Otherwise, I'm going to move this from WIP to full PR in the late morning on the US East Coast tomorrow.

**Comment by wraydulany** [comment]
Ok, changing to full PR. I should note for the record that the previously-working-for-me Pirk Releases also no longer pass the ES and ESS tests for me, another sign that It's Not My Fault.

**Comment by ellisonanne** [comment]
+1 - looks good to merge

File: src/main/java/org/apache/pirk/query/wideskies/Query.java [hunk 4]
```diff
@@ -26,41 +26,51 @@
...
 import org.apache.pirk.serialization.Storable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Class to hold the PIR query vectors
- *
  */
+
 public class Query implements Serializable, Storable
```
Comment (tellison) [inline src/main/java/org/apache/pirk/query/wideskies/Query.java]: I assume we can also drop ```implements Serializable``` now too, right?
This instance, and friends elsewhere.

File: src/main/java/org/apache/pirk/query/wideskies/Query.java [hunk 5]
```diff
@@ -26,41 +26,51 @@
...
+  @Expose private final SortedMap<Integer,BigInteger> queryElements; // query elements - ordered on insertion
 
   // lookup table for exponentiation of query vectors - based on dataPartitionBitSize
   // element -> <power, element^power mod N^2>
   private Map<BigInteger,Map<Integer,BigInteger>> expTable = new ConcurrentHashMap<>();
 
   // File based lookup table for modular exponentiation
   // element hash -> filename containing it's <power, element^power mod N^2> modular exponentiations
-  private Map<Integer,String> expFileBasedLookup = new HashMap<>();
+  @Expose private Map<Integer,String> expFileBasedLookup = new HashMap<>();
```
Comment (tellison) [inline src/main/java/org/apache/pirk/query/wideskies/Query.java]: Should the hash -> filename map be serialized?
AFAIK it is only used by the responder to compute the results, so I don't think it needs serializing.

File: src/main/java/org/apache/pirk/query/wideskies/Query.java [hunk 6]
```diff
@@ -114,8 +124,7 @@ public void generateExpTable()
 
     queryElements.values().parallelStream().forEach(new Consumer<BigInteger>()
     {
-      @Override
-      public void accept(BigInteger element)
+      @Override public void accept(BigInteger element)
```
Comment (tellison) [inline src/main/java/org/apache/pirk/query/wideskies/Query.java]: IDE formatting wars ;-)

**Pull request closed without merging**